modules/compute-vm/variables.tf

```diff
@@ -392,3 +392,12 @@ variable "zone" {
   description = "Compute zone."
   type        = string
 }
+
+variable "gpu" {
+  description = "GPU information."
+  type = object({
+    count = number
+    type  = string
```

Comment:
we typically treat strings that only accept a fixed set of values as enums, and enforce the possible values via avriable validation; check the DNS module for one of many examples